official/v7/entity_response.go

```diff
@@ -159,3 +167,95 @@ func checkErrorFromResponse(response *esapi.Response) error {
 
 	return err
 }
+
+func parseAggregations(
+	aggs map[string]interface{},
+) (ResponseAggregation, error) {
+	const op = errors.Op("parseAggregations")
+
+	var responseAggregation ResponseAggregation
+
+	if aggs == nil {
+		return responseAggregation, nil
+	}
+
+	for aggName, aggData := range aggs {
+		parsedAggData, err := parseAggregationData(aggName, aggData)
+		if err != nil {
+			return responseAggregation, errors.E(op, err)
+		}
+
+		responseAggregation = responseAggregation.AppendAggregation(parsedAggData)
+	}
+
+	return responseAggregation, nil
+}
+
+func parseAggregationData(aggName string, aggData interface{}) (interface{}, error) {
+	const op = errors.Op("parseAggregationData")
+
+	aggMap, ok := aggData.(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("%s: failed to decode aggregation data from response", op)
+	}
+
+	if value, ok := aggMap["value"]; ok {
+		return ResponseMetricAggregation{
+			Name:  aggName,
+			Value: value,
+		}, nil
+	}
+
+	if buckets, ok := aggMap["buckets"].([]interface{}); ok {
+		bucketAggregation := ResponseBucketAggregation{
+			Name: aggName,
+		}
+
+		for _, bucket := range buckets {
+			bucketMap, ok := bucket.(map[string]interface{})
+			if !ok {
+				return nil, errors.Errorf("%s: failed to decode bucket aggregation data from response", op)
+			}
+
+			responseBucket := ResponseBucket{}
+
+			for name, data := range bucketMap {
+				switch name {
```

Comment:
None of these cases are handling unexpected data types. Failing silently could cause confusion for the user. I think we should handle these errors.